docs/proposals/disk-affinity.md

```diff
@@ -0,0 +1,161 @@
+                   name: web
+              volumeMounts:
+                 - name: www-1
+                   mountPath: /usr/share/nginx/html.1
+                 - name: www-2
+                   mountPath: /usr/share/nginx/html.2
+   volumeClaimTemplates:
+      - metadata:
+           name: www-1
+        spec:
+           accessModes: [ "ReadWriteOnce" ]
+           storageClassName: "csi-baremetal-sc-hddlvg"
+           resources:
+              requests:
+                 storage: 1Gi
+      - metadata:
+           name: www-2
+        spec:
+           accessModes: [ "ReadWriteOnce" ]
+           storageClassName: "csi-baremetal-sc-hddlvg"
+           resources:
+              requests:
+                 storage: 1Gi
+
+```
+
+#### Baremetal CSI Driver's improvements:
+
+At Baremetal CSI Driver we should improve Reservation flow (at reservation controller component) for supporting new type of volumes 
+planning (currently we support planning, only depending on capacity). For this we should make the following improvements:
+1. Improve AvailableCapacityReservation CRD. For the improvement we should update AvailableCapacityReservation proto message: \
+1.1. Currently, it's structure is as follows:
+   ```protobuf
+   message AvailableCapacityReservation {
+      string Namespace = 1;
+      string Status = 2;
+      NodeRequests NodeRequests = 3;
+      repeated ReservationRequest ReservationRequests = 4;
+   }
+   ```
+   where NodeRequests - are requests by nodes, ReservationRequests - requests per volumes. \
+1.2. We should update it to support new type of requests: based on affinity annotations:
+```protobuf
+message AvailableCapacityReservation {
+   string Namespace = 1;
+   string Status = 2;
+   NodeRequests NodeRequests = 3;
+   repeated ReservationRequest ReservationRequests = 4;
+   DriveRequests DriveRequests = 5;
+}
+
+message DriveRequests {
+   // affinity requests - filled by scheduler/extender
+   repeated DriveAffinityRequests DriveAffinityRequests = 1;
+   // antiaffinity requests - filled by scheduler/extender
+   repeated DriveAntiaffinityRequests DriveAntiaffinityRequests = 2; 
+}
+
+message DriveAffinityRequests {
+   repeated DriveAffinityRequest DriveAffinityRequest = 1;
+}
+
+message DriveAffinityRequest {
+   DriveAffinityRequestType Type = 1;
+}
+
+enum DriveAffinityRequestType {
+   POD_BOUND = 1;
+} 
+
+message DriveAntiaffinityRequests {
+   repeated DriveAntiaffinityRequest DriveAntiaffinityRequest = 1;
+}
+
+message DriveAntiaffinityRequest {
+   DriveAntiaffinityRequestType Type = 1;
+   oneof Request {
+      DriveAntiaffinityPodLabelRequest PodLabelRequest = 2;
+   }  
+}
+
+enum DriveAntiaffinityRequestType {
+   POD_LABEL = 1;
+}
+
+message DriveAntiaffinityPodLabelRequest {
+  repeated string Labels = 1;
+}
+```
+2. At scheduler extender (in case client set the drive affinity annotations) while creating capacity reservation - fill it with DriveRequests specified in p1.2 above.
```

Comment:
Would it work in case without deploying extender and ACR usage? For example, if we implement Storage Capacity feature. Link - https://kubernetes.io/docs/concepts/storage/storage-capacity/

Reply:
It won't work it this case. But have we already proposals for supporting this feature? If we haven't so - it's better to write it first and then come up with this question.

Reply:
Storage Capacity feature is also planned for implementation and it worth to consider this use case as well.

Reply:
Added section about storage capacity feature to proposal